applications/openshift/authentication/ocp_no_ldap_insecure/policy/stig/shared.yml

```diff
@@ -0,0 +1,139 @@
+srg_requirement: |-
+  {{{ full_name }}} must use a centralized user management solution to
+  support account management functions.
+
+vuldiscussion: |-
+  Enterprise environments make application account management challenging
+  and complex. A manual process for account management functions adds the
+  risk of a potential oversight or other error.
+
+checktext: |-
+  Verify that the authentication operator is configure to use either an
+  LDAP or a OpenIDConnect or an approved identity provider:
+
+  > oc get oauth cluster -o
+  jsonpath="{.spec.identityProviders[*].type}{'\n'}"
+
+  If the output lists HTPasswd, this is
```

Comment:
I think this looks good here

Reply:
Oh you were right and I realized I didn't provide good enough explanation. So on one hand, since this control is fixed by four different rules, each of the rules and their policy files should contain mostly description that pertains to that rule only - so this rule should mostly talk about checking that if LDAP is used, then the insecure flag is not used. On the other hand, the fix, at least when it comes to setting up the IDP is kindof generic.

I changed the checks of the rules to only talk about what they are supposed to do and left the IDP related checks mostly generic. The kubeadmin removed check and fix can be more targeted.

btw the way I was checking the result was:

1. build the content - `./build_product --debug ocp4`
2. build the rules JSON - `export PYTHONPATH=$(pwd); python3 ./utils/rule_dir_json.py`
3. build the HTML version of the export - `export PYTHONPATH=$(pwd); python3 utils/create_srg_export.py -c controls/srg_ctr.yml -p ocp4 -m shared/references/disa-ctr-srg-v1r3.xml --out-format html --output /tmp/srg-mapping-ocp4.html`
4. compare the individual controls with what is at https://docs.google.com/spreadsheets/d/1oTUEUSkpumEpGqeBvhEHkJxcGUxMgoG1/edit#gid=959715005 (I made a copy so that I don't accidentally edit this document)

Reply:
Thanks for explaining this here!